src/bridge-deployer/src/contracts.rs

```diff
@@ -332,26 +335,65 @@ impl SolidityContractDeployer<'_> {
         Ok(contract_address)
     }
 
-    /// Returns the nonce of the deployer.
-    pub async fn get_nonce(&self) -> Result<u64> {
+    fn rpc_client(&self) -> anyhow::Result<EthJsonRpcClient<ReqwestClient>> {
         let url = self.get_network_url();
-
-        debug!("Requesting nonce with EVM url: {url}");
-
         let reqwest_client = reqwest::ClientBuilder::new()
             .danger_accept_invalid_certs(true)
             .build()?;
         let client = EthJsonRpcClient::new(ReqwestClient::new_with_client(
             url.to_string(),
             reqwest_client,
         ));
+
+        Ok(client)
+    }
+
+    /// Returns the nonce of the deployer.
+    pub async fn get_nonce(&self) -> Result<u64> {
+        let network = self.get_network_url();
+        info!("Requesting nonce for network {network}");
+
+        let client = self.rpc_client()?;
         let address = self.wallet.address();
         let nonce = client
             .get_transaction_count(address, BlockNumber::Latest)
             .await?;
 
-        info!("Got nonce value: {nonce}");
+        info!("Got nonce value for network {network}: {nonce}");
 
         Ok(nonce)
     }
+
+    pub async fn transfer_eth(&self, to: &H160, amount: u128) -> Result<()> {
+        info!(
+            "Transferring {amount} ETH tokens to address {}",
+            to.encode_hex_with_prefix()
+        );
+        let client = self.rpc_client()?;
+
+        let address = self.wallet.address();
+        let nonce = client
+            .get_transaction_count(address, BlockNumber::Latest)
+            .await?;
+        let chain_id = client.get_chain_id().await?;
+
+        let tx = TransactionBuilder {
+            from: &address.into(),
+            to: Some((*to).into()),
+            nonce: nonce.into(),
+            value: amount.into(),
+            gas: 5_000_000u64.into(),
```

Comment:
Isn't it 5M a lot?

Reply:
As far as I understand thsi option is just the limit we are willing to spend on the transaction. Actual value used will be calculated by the network.